libraries/commerce/cart/actions/addProductsToCart.js

```diff
@@ -6,43 +6,58 @@ import addProductsToCart from '../action-creators/addProductsToCart';
 import successAddProductsToCart from '../action-creators/successAddProductsToCart';
 import errorAddProductsToCart from '../action-creators/errorAddProductsToCart';
 import setCartProductPendingCount from '../action-creators/setCartProductPendingCount';
-import { getProductPendingCount } from '../selectors';
+import { getProductPendingCount, getAddToCartMetadata } from '../selectors';
+import { getProductById } from '../../product/selectors/product';
 import { messagesHaveErrors } from '../helpers';
 
 /**
- * Adds a product to the cart.
+ * Adds a product to the cart. Includes metadata if available and not provided before.
  * @param {Array} productData The options for the products to be added.
  * @return {Function} A redux thunk.
  */
 const addToCart = productData => (dispatch, getState) => {
-  const pendingProductCount = getProductPendingCount(getState());
+  const state = getState();
+  const pendingProductCount = getProductPendingCount(state);
+  const products = productData.map(product => {
+    const { productData } = getProductById(state, product.productId);
+    const productId = productData.baseProductId || product.productId;
 
-  dispatch(addProductsToCart(productData));
+    // it's a variant if it has a baseProductId
+    const variantProductId = productData.baseProductId ? productData.id : undefined;
+    const metadata = getAddToCartMetadata(state, productId, variantProductId);
```

Comment:
This will not work when, we "deeplink" to the favorites page: in that moment we have the child product in the redux store but the parent is not. 
Clicking on addToCart now will throw the error `Cannot read property 'productData' of undefined`. We need to adjust the getProductMetadata selector to check for product being empty.

Reply:
Please check eslint codestyle

Reply:
fixed the error `Cannot read property 'productData' of undefined`